packages/charts/src/scales/types.ts

```diff
@@ -23,7 +23,7 @@ export function isLogarithmicScale(scale: Scale<unknown>): scale is ScaleContinu
  * Check if a scale is Band
  * @internal
  */
-export function isBandScale(scale: Scale<unknown>): scale is ScaleBand {
+export function isBandScale(scale: Scale<unknown>): scale is ScaleBand<string | number> {
```

Comment:
I think something like this would work a little nicer and automatically infer the type of `T`

```suggestion
export function isBandScale<T extends string | number>(scale: Scale<T>): scale is ScaleBand<T> {
```

Reply:
Thanks, good tip! Held off on turning functions generic, in an attempt to first just remove the top level `any` domain

Reply:
Right but doing so using `as` could create unexpected runtime errors and making the function generic with default covers both cases.

Reply:
💯 % agree, committing your suggestion

Reply:
8cd6cda702ab6ee1982c59076b6bfa7e18d7f8c3